catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -0,0 +1,390 @@
+        queryExecutor = Executors.newSingleThreadExecutor();
+
+        preQueryPlugin = mock(PreFederatedQueryPlugin.class);
+        when(preQueryPlugin.process(any(),
+                any())).thenAnswer(invocation -> invocation.getArguments()[1]);
+
+        strategy = new CachingFederationStrategy(queryExecutor,
+                Arrays.asList(preQueryPlugin),
+                new ArrayList<>(),
+                new SolrCache(mock(SolrClientAdaptor.class)),
+                executor,
+                mock(ValidationQueryFactory.class),
+                new CacheQueryFactory(new GeotoolsFilterBuilder())) {
+            protected Runnable createMonitor(
+                    final CompletionService<SourceResponse> completionService,
+                    final Map<Future<SourceResponse>, QueryRequest> futures,
+                    final QueryResponseImpl returnResults, final QueryRequest request) {
+
+                return new SortedQueryMonitor(this,
+                        completionService,
+                        futures,
+                        returnResults,
+                        request,
+                        postQuery) {
+                    @Override
+                    public void run() {
+                        returnResults.setHits(42);
+                    }
+                };
+            }
+        };
+
+        mockQuery = mock(Query.class);
+        when(mockQuery.getTimeoutMillis()).thenReturn(LONG_TIMEOUT);
+        when(mockQuery.getPageSize()).thenReturn(-1);
+
+        refreshExecutor();
+    }
+
+    @After
+    public void tearDownClass() throws Exception {
+        strategy.shutdown();
+        killAndWaitForExecutor();
+    }
+
+    @Test
+    public void testFederateQueryCache() throws Exception {
+
+        HashMap<String, Serializable> properties = new HashMap<>();
+        properties.put(QUERY_MODE, CACHE_QUERY_MODE);
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        MetacardImpl mockMetacard = new MetacardImpl();
+
+        Result mockResult = new ResultImpl(mockMetacard);
+
+        SourceResponse mockResponse = mock(SourceResponse.class);
+        List<Result> results = Arrays.asList(mockResult);
+        when(mockResponse.getHits()).thenReturn((long) results.size());
+        when(mockResponse.getResults()).thenReturn(results);
+
+        MockSource mockSource = new MockSource("mock source",
+                "mock title",
+                "mock 1.0",
+                "mock org",
+                new HashSet<>(),
+                true,
+                new Date());
+
+        mockSource.setQueryResponse(mockResponse);
+
+        List<Source> sources = new ArrayList<>();
+        sources.add(mockSource);
+
+        QueryResponse response = strategy.federate(sources, fedQueryRequest);
+
+        WaitCondition.expect("Wait until the result appears in the response")
+                .until(() -> response.getHits(), is(42L))
```

Comment:
I recommend making the result hits a constant so if it ever changes, we only have to change it in one place